packages/realm-react/src/cachedObject.ts

```diff
@@ -0,0 +1,83 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import Realm from "realm";
+import { cachedCollection } from "./cachedCollection";
+
+type IndexibleRealmObject = Realm.Object & { [key: string]: unknown };
+
+export function cachedObject<T extends Realm.Object>(
+  object: T | null,
+  updateCallback: () => void,
+  listCaches = new Map(),
+  listTeardowns: (() => void)[] = [],
```

Comment:
Nitpick, but I guess we should be consistent!
```suggestion
  listTearDowns: (() => void)[] = [],
```

Reply:
But of course!  Good catch